qiita_db/metadata_template/test/test_sample_template.py

```diff
@@ -1407,72 +1331,63 @@ def test_extend_add_samples(self):
 
         npt.assert_warns(QiitaDBWarning, st.extend, md_ext)
 
-        # Test samples were appended successfully to the required sample info
-        # table
-        study_id = self.new_study.id
-        sql = """SELECT *
-                 FROM qiita.study_sample
-                 WHERE study_id=%s"""
-        obs = [dict(o)
-               for o in self.conn_handler.execute_fetchall(sql, (study_id,))]
-        exp = [{'sample_id': '2.Sample1',
-                'study_id': 2},
-               {'sample_id': '2.Sample2',
-                'study_id': 2},
-               {'sample_id': '2.Sample3',
-                'study_id': 2},
-               {'sample_id': '2.Sample4',
-                'study_id': 2},
-               {'sample_id': '2.Sample5',
-                'study_id': 2}]
-        self.assertItemsEqual(obs, exp)
-
-        # Test samples were appended successfully to the dynamic table
-        sql = "SELECT * FROM qiita.sample_{0}".format(study_id)
-        obs = [dict(o) for o in self.conn_handler.execute_fetchall(sql)]
-        exp = [{'sample_id': '2.Sample1',
-                'int_column': 1,
-                'str_column': 'Value for sample 1',
-                'physical_specimen_location': 'location1',
-                'physical_specimen_remaining': True,
-                'dna_extracted': True,
-                'sample_type': 'type1',
+        # Test samples have been added correctly
+        exp_sample_ids = {"%s.Sample1" % st.id, "%s.Sample2" % st.id,
+                          "%s.Sample3" % st.id, "%s.Sample4" % st.id,
+                          "%s.Sample5" % st.id}
+        self.assertEqual(st._get_sample_ids(), exp_sample_ids)
+        self.assertEqual(len(st), 5)
+        exp_categories = {'collection_timestamp', 'description',
+                          'dna_extracted', 'host_subject_id', 'int_column',
+                          'latitude', 'longitude',
+                          'physical_specimen_location',
+                          'physical_specimen_remaining', 'sample_type',
+                          'scientific_name', 'str_column', 'taxon_id'}
+        self.assertEqual(set(st.categories()), exp_categories)
```

Comment:
Same as above, I will stop marking these in case we reach a resolution that's not applicable for the general case.